.github/workflows/binary-applications.yml

```diff
@@ -210,9 +199,10 @@ jobs:
               if: github.event_name != 'pull_request'
               shell: bash -l {0}
               env:
-                CERT_KEY_PASS: ${{ secrets.STANFORD_CERT_KEY_PASS }}
+                CERT_FILE: 2025-01-16-Expiry-AppStore-App.p12
+                CERT_PASS: ${{ secrets.MACOS_CODESIGN_CERT_PASS }}
                 KEYCHAIN_PASS: ${{ secrets.KEYCHAIN_PASS }}

```

Comment:
I'm not sure whether it's necessary to keep the keychain password a secret -  I'm leaning towards no. If someone had accessed the github actions VM and could use the keychain password, then they'd also have access to the certificate password. I don't think it adds any more security.

Reply:
Sorry I'm not very plugged in to this topic, what is the keychain password? Knowing pretty little about this my initial questions are:
- Is this something that would change often?
- Is there a downside from keeping it a secret?
- If someone cloned our repo without the keychain password being a secret to do what they wish with it, would that pose an issue?

Reply:
On macos, signing certificates need to be accessed through a keychain, which is the password/secret management system. A keychain can contain passwords, signing identities, etc. And keychains themselves are password-protected.

There's a default keychain, but we can't access it because the password is the user login password, which we can't know on Github Actions. We have to create a new keychain with a known password so we can unlock it and import the signing certificate.

No other secrets will be stored in the new keychain, and the signing certificate is still protected by its own password. Someone would have to (1) get the certificate file and (2) get the corresponding password to do anything malicious with it.

Reply:
* It would not change often, it's a totally arbitrary password that we make up just so we can create the keychain.
* The downside of keeping it secret is just that it complicates things slightly and we have to store it as a github secret in all forks.

Reply:
@emlys I think I'm convinced it doesn't seem necessary to keep as a GHA secret. Thanks for taking time to write up that rundown.